### PR TITLE
ssh key validation

### PR DESCRIPTION
**Pull request by sheldor1510**
- ssh key validation done with `PublicKeyLoader`
- added ajax call for ssh key validation on textarea change
- added red box shadow to textarea and disabled "Add Key" button if key is not valid
- added ssh key validation code on the backend for both cases of paste and file import
- #17

### CODE REVIEW

File: webroot/panel/account.php [hunk 1]
```diff
@@ -3,20 +3,37 @@
...
 
             switch ($_POST["add_type"]) {
                 case "paste":
-                    array_push($added_keys, $_POST["key"]);
+                    try {
+                        PublicKeyLoader::load($_POST["key"], $password = false);
+                        array_push($added_keys, $_POST["key"]);
+                    } catch (Exception $e) {
+                        echo $invalid_ssh_dialogue;
+                    }
```
Comment (hakasapl) [inline webroot/panel/account.php:26]: I'd prefer if this block of code is written in a function on the top of this file, then used twice in the code up here. Something like `testValidSSHKey($key_str)` which returns true/false

File: webroot/panel/account.php [hunk 2]
```diff
@@ -3,20 +3,37 @@
 require_once "../../resources/autoload.php";
 
 use UnityWebPortal\lib\UnitySite;
+use phpseclib3\Crypt\PublicKeyLoader;
 
 require_once $LOC_HEADER;
 
+$invalid_ssh_dialogue = "<script type='text/javascript'>
+alert('Invalid SSH Key. Try again.');
```
Comment (hakasapl) [inline webroot/panel/account.php:11]:
```suggestion
alert('Invalid SSH key. Please verify your public key file is valid.');
```

**Review by hakasapl — CHANGES_REQUESTED** [review]
Some minor code cleanup changes, other than that this looks really good, thanks!

File: webroot/panel/account.php (hunk 1 above)
Comment (sheldor1510) [reply webroot/panel/account.php]: Added the function but can't fix one codesniffer warning

**Comment by sheldor1510** [comment]
@hakasapl can you fix this one codesniffer warning?

`A file should declare new symbols (classes, functions, constants, etc.) and cause no other side effects, or it should execute logic with side effects, but should not do both. The first symbol is defined on line 5 and the first side effect is on line 15 (PSR1.Files.SideEffects.FoundWithSymbols)`

**Comment by hakasapl** [comment]
Hi, it's complaining because the function can't be in this file. Could you make that a static function and put it in this file? https://github.com/UnityHPC/unity-web-portal/blob/main/resources/lib/UnitySite.php

That should resolve that issue, then you'll have to call the static function from the account page

**Pull request merged**